src/libraries/System.Private.CoreLib/src/System/Type.cs

```diff
@@ -158,17 +185,27 @@ public virtual Type[] GetGenericParameterConstraints()
         public MemberInfo[] GetMembers() => GetMembers(Type.DefaultLookup);
         public abstract MemberInfo[] GetMembers(BindingFlags bindingAttr);
 
+        [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods)]
```

Comment:
Maybe I'm not understanding how this is supposed to work.  Does DynamicallyAccessedMembers in this case cause the linker to keep all of the public methods of the named type?  I'd have thought the linker would know about Type.GetMethod(string), and would only keep alive methods that matched that name.  If it does do that, why is the DynamicallyAccessedMembers necessary, just to suppress linker warnings?  Does it not keep additional methods alive unnecessarily?

Reply:
Linker intrinsically recognizes methods like this and if it sees the method name/binding flags/parameters statically, it will do the right thing.

This is annotated so that potentially other tools that don't recognize this as intrinsic can still be correct (even though they won't produce the most efficient output).